packages/perspective-viewer-d3fc/src/js/charts/heatmap.js

```diff
@@ -0,0 +1,76 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2017, the Perspective Authors.
+ *
+ * This file is part of the Perspective library, distributed under the terms of
+ * the Apache License 2.0.  The full license can be found in the LICENSE file.
+ *
+ */
+import * as d3 from "d3";
+import * as fc from "d3fc";
+import * as crossAxis from "../axis/crossAxis";
+//import * as otherAxis from "../axis/otherAxis";
+import {heatmapSeries} from "../series/heatmapSeries";
+import {heatmapData} from "../data/heatmapData";
+import {filterData} from "../legend/legend";
+import {getOrCreateElement} from "../utils/utils";
+import scrollableLegend from "../legend/scrollableLegend";
+import chartSvgCartesian from "../d3fc/chart/svg/cartesian";
+import {withGridLines} from "../gridlines/gridlines";
+
+function heatmapChart(container, settings) {
+    const data = heatmapData(settings, filterData(settings));
+    const colorInterpolate = d3.interpolateViridis;
+
+    const series = heatmapSeries(settings, colorInterpolate);
+
+    const uniqueYDomain = [...new Set(data.map(d => d.mainValue))];
+    const extent = fc.extentLinear().accessors([d => d.colorValue]);
+
+    const colourDomain = extent(data);
+    legend(container, colorInterpolate, colourDomain);
+
+    const chart = chartSvgCartesian(crossAxis.scale(settings), crossAxis.scale(settings, "splitValues"))
+        .xDomain(crossAxis.domain(settings)(settings.data))
+        .yDomain(uniqueYDomain)
```

Comment:
It seems like we should be able to do something like: `crossAxis.domain(settings).valueName('mainValue')(settings.data)` like we do with the `mainAxis` in x-y.

Reply:
@DevAndyLee 
We could do something similar, but changing the existing crossAxis.domain function would be tricky as it already has lots of logic in it and relies quite heavily on using labelFunction which doesn't apply in this context.
 Would a new function be OK?
E.g. crossAxis.domainFromData(settings).valueName('mainValue')(data)
Ideally the domain values should come from the re-organised data structure rather than from settings.data as it is so much each (like we do for mainAxis in x-y)